dashboard/app/controllers/api/v1/regional_partners_controller.rb

```diff
@@ -77,4 +92,40 @@ def get_partner_cohort_capacity(regional_partner_value, role)
     end
     nil
   end
+
+  # Get the regional partner's number of enrollments to current-year role-specific workshops.
+  #
+  # @param regional_partner_id [Integer]
+  # @param role [String] a string contains workshop course and subject type, e.g. 'csp_teachers'
+  # @return [Integer, nil] nil if regional_partner_id is invalid, otherwise returns a count
+  # @see Api::V1::Pd::ApplicationsController:ROLES
+  #
+  def get_partner_enrollment_count(regional_partner_id, role)
+    partner = RegionalPartner.find_by_id(regional_partner_id)
+    return unless partner
+
+    workshops = partner.pd_workshops
+    workshops_by_role =
+      case role.to_sym
+      when :csf_facilitators
+        workshops.where(course: COURSE_CSF, subject: SUBJECT_CSF_FIT)
+      when :csd_facilitators
+        workshops.where(course: COURSE_CSD, subject: SUBJECT_CSD_FIT)
+      when :csp_facilitators
+        workshops.where(course: COURSE_CSP, subject: SUBJECT_CSP_FIT)
+      when :csd_teachers
+        workshops.where(course: COURSE_CSD).where.not(subject: SUBJECT_CSD_FIT)
+      when :csp_teachers
+        workshops.where(course: COURSE_CSP).where.not(subject: SUBJECT_CSP_FIT)
+      else
+        # Should never get here. This value will cause an exception on the next calculation,
+        # which expects an ActiveRecord::Relation.
+        nil
+      end
+
+    workshop_current_year_ids =
+      workshops_by_role.select {|ws| ws.school_year == APPLICATION_CURRENT_YEAR}.pluck(:id)
+
+    Pd::Enrollment.where(pd_workshop_id: [workshop_current_year_ids]).count
```

Comment:
How long does this query take? Wondering if it runs into the same mysql query optimizer bug that Will worked around in https://github.com/code-dot-org/code-dot-org/pull/31193.

Reply:
Wow I don't know about Will's PR. It's helpful! I assume it is for https://app.honeybadger.io/projects/3240/faults/55727995. 

The query here is instantly for the regional partner with the most workshops, 116. This query just counts number of rows in 1 table so I don't think it will hit the same bug.